### PR TITLE
Update local volume docs with version compatibility, reorg known issues

### PR DESCRIPTION
**Pull request by msau42**
Trying to differentiate between K8s and provisioner versions.

### CODE REVIEW

File: local-volume/KNOWN_ISSUES.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+### 1.0: 
+* The provisioner will not correctly detect new mounts added after it has been started
+    * Fixed in provisioner 2.0 when the mount propagation
+      alpha feature gate is enabled, which is available in Kubernetes 1.8+.
+    * The provisioner 1.x will detect the existance of a new directory in the
+      discovery directory. Then, it will incorrectly create a local PV with
+      the root filesystem capacity.
+    * 1.x workaround: Before adding any new mount points, stop
+      the provisioner daemonset, add the new mount points, start the daemonset.
```
Comment (ianchakeres) [inline local-volume/KNOWN_ISSUES.md]: nit: /start the daemonset/restart the daemonset/

File: local-volume/README.md [hunk 2]
```diff
@@ -22,66 +22,43 @@ directories by creating and cleaning up PersistentVolumes for each volume.
...
+* Dynamic provisioning for shared local persistent storage
 * Local PV health monitoring, taints and tolerations
 * Inline PV (use dedicated local disk as ephemeral storage)
-* Dynamic provisioning for shared local persistent storage
+
+## Provisioner Compatibility
+
+| Provisioner Version | Min K8s Version |
+| ------------------- | --------------- |
+| 2.0.0               | 1.8             |
```
Comment (ianchakeres) [inline local-volume/README.md]: Why can't 2.0 provisioner be used with 1.7?

Comment (msau42) [reply local-volume/README.md]: Oh it's if you want to fix the mount propagation issue.  I guess it's strictly not required if you don't care about it.  Hmmm this is becoming more difficult to convey.

I imagine we'll want to say things like:
- If you want to fix the mount propgation issue, then you must use K8s 1.8 + provisioner 2.0.0
- If you want block volume support, then you must use K8s 1.10 + provisioner 2.0.1
- If you want to use beta local volume API, then you must use K8s 1.10 + provisioner 2.1.0
- When alpha local volume API is deprecated and removed, then you can't use provisioner < X.

Some dependencies are soft (ie only if you need that feature), and some may be hard (ie alpha API will be deprecated and removed, so lower versions will definitely not work anymore, and higher versions won't support alpha API).  

I am open to suggestions on how to convey this nicely.

Comment (ianchakeres) [reply local-volume/README.md]: I agree that it is hard to describe all the various potential configurations and combinations.

I'm inclined to recommend that people use the latest version of kubernetes and the latest version of the provisioner with all the listed feature-gates, and that we describe the latest version's functions and capabilities here. 

We can point to CHANGELOG for bugfixes and addition of new features (e.g. block). We can point to KNOWN_ISSUES to explain earlier version issues, and mention which provisioner-version/k8s-version/feature-gate combination addresses a particular issue earlier.

Comment (msau42) [reply local-volume/README.md]: K I'm trying this out:
Provisioner features are in provisioner CHANGELOG.  Known issues are in KNOWN_ISSUES.
K8s features are in main README. Known issues are in KNOWN_ISSUES.

**Pull request merged**